Realm/RLMApp.mm

```diff
@@ -0,0 +1,178 @@
+                      transport:(id<RLMNetworkTransporting> _Nullable)transport
+                   localAppName:(NSString * _Nullable)localAppName
+                localAppVersion:(NSString * _Nullable)localAppVersion
+        defaultRequestTimeoutMS:(NSUInteger)defaultRequestTimeoutMS {
+    if (self = [super init]) {
+        self.baseURL = baseURL;
+        self.transport = transport;
+        self.localAppName = localAppName;
+        self.localAppVersion = localAppVersion;
+        self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
+        return self;
+    }
+    return nil;
+}
+@end
+
+/// A map of ObjectStore HTTP methods to RLMHTTPMethods
+std::map<app::HttpMethod, RLMHTTPMethod> HttpMethod_toRLMHTTPMethod = {
+    {app::HttpMethod::get, GET},
+    {app::HttpMethod::post, POST},
+    {app::HttpMethod::put, PUT},
+    {app::HttpMethod::patch, PATCH},
+    {app::HttpMethod::del, DELETE}
+};
+
+/// Internal transport struct to bridge RLMNetworkingTransporting to the GenericNetworkTransport.
+struct CocoaNetworkTransport : public realm::app::GenericNetworkTransport {
+    static id <RLMNetworkTransporting> transport;
+
+    void send_request_to_server(const app::Request request,
+                                std::function<void(const app::Response)> completion) override {
+        // Convert the app::Request to an RLMRequest
+        auto rlmRequest = [RLMRequest new];
+        rlmRequest.url = @(request.url.data());
+        rlmRequest.body = @(request.body.data());
+        NSMutableDictionary *headers = [NSMutableDictionary new];
+        for (auto header : request.headers) {
+            headers[@(header.first.data())] = @(header.second.data());
+        }
+        rlmRequest.headers = headers;
+        rlmRequest.method = HttpMethod_toRLMHTTPMethod[request.method];
+        rlmRequest.timeoutMS = request.timeout_ms;
+
+        // Send the request through to the Cocoa level transport
+        [transport sendRequestToServer:rlmRequest completion:^(RLMResponse * _Nonnull response) {
+            std::map<std::string, std::string> bridgingHeaders;
+            for (id key in response.headers) {
+                bridgingHeaders[[key cStringUsingEncoding:NSUTF8StringEncoding]] = [response.headers[key] cStringUsingEncoding:NSUTF8StringEncoding];
+            }
+
+            // Convert the RLMResponse to an app:Response and pass downstream to
+            // the object store
+            completion(app::Response {
+                .body = [response.body cStringUsingEncoding:NSUTF8StringEncoding],
+                .headers = bridgingHeaders,
+                .http_status_code = static_cast<int>(response.httpStatusCode),
+                .custom_status_code = static_cast<int>(response.customStatusCode)
+            });
+        }];
+    }
+};
+
+id <RLMNetworkTransporting> CocoaNetworkTransport::transport = [RLMNetworkTransport new];
+
+@implementation RLMApp : NSObject
+
++(NSMutableDictionary<NSString *, RLMApp*>*)apps {
```

Comment:
This probably should be a free function and not a method on RLMApp? It doesn't need anything from RLMApp and shouldn't be exposed as part of RLMApp's interface.